package.json

```diff
@@ -32,20 +35,29 @@
     "gulp-rename": "^2.0.0",
     "gulp-replace": "^1.0.0",
     "gulp-sourcemaps": "^3.0.0",
-    "gulp-tslint": "^8.1.4",
     "gulp-typedoc": "^2.2.5",
     "gulp-typescript": "^5.0.1",
     "gulp-uglify": "^3.0.2",
+    "husky": "^4.3.0",
     "lerna": "^3.22.0",
+    "lint-staged": "^10.5.1",
     "merge2": "^1.4.1",
     "remap-istanbul": "^0.13.0",
     "rollup": "^2.13.0",
     "rollup-plugin-sourcemaps": "^0.6.2",
     "rollup-plugin-typescript": "^1.0.1",
     "run-sequence": "^2.2.1",
-    "tslint": "^5.20.1",
-    "tslint-microsoft-contrib": "^6.2.0",
     "typedoc": "^0.19.2",
     "typescript": "^3.9.4"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "lint-staged"
+    }
+  },
+  "lint-staged": {
+    "*.{js,ts,tsx}": [
```

Comment:
should this just be . like in the npm scripts above?

Reply:
in this case I think the way lint-staged works is it passes all files that match that wildcard to eslint. So if we were to change it to (`.` or I think it would actually be `*`) it would send all file types to eslint which would cause it to error.

Reply:
you can see for yourself this behaviour by running `eslint README.md`